sbi/utils/sbiutils.py

```diff
@@ -232,6 +239,9 @@ def handle_invalid_x(
 
     batch_size = x.shape[0]
 
+    # Squeeze to cover all dimensions in case of multidimensional x.
+    x = x.reshape(batch_size, -1)
```

Comment:
I'm confused. In the warning above, you write that `sbi` can not deal with multi-d `x`. But here, it seems as if we automatically flattened which would then work, no?

Reply:
The above warning says that with **default settings** it won't work. 

Note that `x` is not returned. We flatten `x` only inside this function in order to allow for multi-d `x` in case of an appropriate embedding. 
We need this flattening to move all `x` into the second dim for the NaN / inf check below.

Reply:
Ah, I missed that this only happens within `handle_invalid_x`. Makes sense then :)